src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResource.kt

```diff
@@ -0,0 +1,122 @@
+package uk.gov.justice.digital.hmpps.prisonregister.resource
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.prisonregister.ErrorResponse
+import uk.gov.justice.digital.hmpps.prisonregister.service.AuditService
+import uk.gov.justice.digital.hmpps.prisonregister.service.AuditType.PRISON_REGISTER_ADDRESS_UPDATE
+import uk.gov.justice.digital.hmpps.prisonregister.service.PrisonAddressService
+import uk.gov.justice.digital.hmpps.prisonregister.service.SnsService
+import java.time.Instant
+import javax.validation.Valid
+import javax.validation.constraints.Size
+
+@RestController
+@Validated
+@RequestMapping(name = "Prison Maintenance", path = ["/prison-maintenance"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class PrisonAddressMaintenanceResource(
+  private val addressService: PrisonAddressService,
+  private val snsService: SnsService,
+  private val auditService: AuditService
+) {
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_REF_DATA') and hasAuthority('SCOPE_write')")
+  @Operation(
+    summary = "Update specified address details",
+    description = "Updates address information, role required is MAINTAIN_REF_DATA",
+    security = [SecurityRequirement(name = "MAINTAIN_REF_DATA", scopes = ["write"])],
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = UpdateAddressDto::class)
+        )
+      ]
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Address Information Updated",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = AddressDto::class))]
```

Comment:
Apparently you don't need this content line

Reply:
Will pull this out in the next PR